### DISCUSSION TITLE
(401) Unauthorized after 1 hour of ApplyProvisioningTemplate

### DISCUSSION THREAD

**Discussion by PedroMordeP** [pnp/pnpframework#412] [Q&A] [answered]
Hello,

I did some research on this and found out the context session is only 1 hour valid and I need to get a new access token.
I have something like this 

static readonly AuthenticationManager auth = new AuthenticationManager();

```
accessToken = await AuthHelper.GetUserAuthorization(authUri, o365User, request.Context.AuthenticationKey, tenantUrl);

using (ClientContext ctx = auth.GetAccessTokenContext(url, accessToken))
{
....
ProvisioningTemplateApplyingInformation ptai = new ProvisioningTemplateApplyingInformation
{
    PersistTemplateInfo = true,
    ClearNavigation = true,
    ProvisionContentTypesToSubWebs = true,
    ProgressDelegate = async delegate (String message, Int32 progress, Int32 total)
    {
         loggerService.LogInformation(string.Format("PROVISIONING - {0:00}/{1:00} - {2}", progress, total, message));
     },
    MessagesDelegate = delegate (String message, ProvisioningMessageType type)
    {
         loggerService.LogInformation(string.Format("PROVISIONING - {0} Type - {1}", message, type), BTProvisioningCode.INSTALLING);
     }
};

ctx.Web.ApplyProvisioningTemplate(template, ptai);
}
```
My question is how I can get a new access token when the deployment is running?
I can't find any examples or mention this on documentation.

Using PnP.FRamework 1.4.0

**Comment by jansenbe** [comment, marked as answer]
Seems like you're getting the access token using your own logic and then provide it...this approach indeed will not automatically refresh the token. If you directly use AuthenticationManager then most methods under covers use MSAL and MSAL will then handle the token refresh for you.
